src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -2448,7 +2448,19 @@ private void LogDisabledPropertyUse(string disabledPropertyName, string newPrope
         public int DatabaseStatementCacheCapcity => _databaseStatementCacheCapcity ?? (_databaseStatementCacheCapcity =
             TryGetAppSettingAsIntWithDefault("SqlStatementCacheCapacity", DefaultSqlStatementCacheCapacity)).Value;
 
-        public bool CodeLevelMetricsEnabled =>  _localConfiguration.codeLevelMetrics.enabled;
+        private bool? _codeLevelMetricsEnabled;
+        public bool CodeLevelMetricsEnabled
+        {
+            get
+            {
+                if (!_codeLevelMetricsEnabled.HasValue)
+                {
+                    _codeLevelMetricsEnabled = EnvironmentOverrides(_localConfiguration.codeLevelMetrics.enabled, "NEW_RELIC_CODE_LEVEL_METRICS_ENABLED");
```

Comment:
I'm not sure about this pattern. While it does reduce the frequency of environment variables being checked, I have a sneaking suspicion it may prevent the configured value from ever being reloaded once the agent starts. 

Will this work when the agent detects a config file change, and reloads?

Reply:
I forgot the agent did config file reloading. I believe there are other configuration items that do similar but that doesn't seem ideal if that's a supported use case.

That being said, we also don't want to be processing the ENV string for every single span event. Really, you only want to be processing ENV strings once on configuration load.

Reply:
Unsure what sort of testing we have around this... so I'll poke around there but here's the code paths.

When a config change is detected via the `ConfigurationTracker` a new instance of `InternalConfiguration` which inherits from `DefaultConfiguration` is created. So the storing in a member would not be a concern for that specific use case.

This would of course not automatically pick up ENV var changes but seems like a stretch to want to support that. If your agent does attempt to support that then that's a whole bag of worms.

https://github.com/newrelic/newrelic-dotnet-agent/blob/main/src/Agent/NewRelic/Agent/Core/Config/ConfigurationTracker.cs#L40

https://github.com/newrelic/newrelic-dotnet-agent/blob/main/src/Agent/NewRelic/Agent/Core/Config/ConfigurationLoader.cs#L364

https://github.com/newrelic/newrelic-dotnet-agent/blob/main/src/Agent/NewRelic/Agent/Core/Configuration/ConfigurationService.cs#L100